plugins/rbac-backend/src/audit-log/audit-logger.ts

```diff
@@ -0,0 +1,323 @@
+import {
+  AuthorizeResult,
+  PolicyDecision,
+  ResourcePermission,
+} from '@backstage/plugin-permission-common';
+import { PolicyQuery } from '@backstage/plugin-permission-node';
+
+import { AuditLogOptions } from '@janus-idp/backstage-plugin-audit-log-common';
+import {
+  PermissionAction,
+  PermissionInfo,
+  RoleConditionalPolicyDecision,
+  Source,
+  toPermissionAction,
+} from '@janus-idp/backstage-plugin-rbac-common';
+
+import { RoleMetadataDao } from '../database/role-metadata';
+import { policyToString, stringToPolicy } from '../helper';
+
+export enum RoleEvents {
+  CreateRole = 'CreateRole',
+  UpdateRole = 'UpdateRole',
+  DeleteRole = 'DeleteRole',
+}
+export enum PermissionEvents {
+  CreatePermission = 'CreatePermission',
+  UpdatePermission = 'UpdatePermission',
+  DeletePermission = 'DeletePermission',
+}
+export type Operation = 'CREATE' | 'UPDATE' | 'DELETE';
+
+export type RoleMembersDiff = {
+  added?: string[];
+  removed?: string[];
+  updated?: string[];
+};
+export type PermissionDiff = {
+  added?: string[][];
+  removed?: string[][];
+  updated?: string[][];
+};
+
+export type RoleAuditInfo = {
+  roleEntityRef: string;
+  description?: string;
+  membersDiff: RoleMembersDiff;
+  source: Source;
+  operation: Operation;
+};
+
+export type PermissionAuditInfo = {
+  policies: string[][];
+  permissionDiff: PermissionDiff;
+  source: Source;
+  operation: Operation;
+};
+
+export enum EvaluationEvents {
+  StartedPermissionEvaluation = 'StartedPermissionEvaluation',
+  EvaluationPermissionCompleted = 'EvaluationPermissionCompleted',
+  EvaluationConditionCompleted = 'EvaluationConditionCompleted',
+}
+
+export type EvaluationAuditInfo = {
+  userEntityRef: string;
+  permissionName: string;
+  action: PermissionAction;
+  resourceType?: string;
+  decision?: PolicyDecision;
+};
+
+export enum ConditionEvents {
```

Comment:
FYI, enums are considered harmful in TS. https://www.youtube.com/watch?v=jjMbPt_H3RQ